app/mender.go

```diff
@@ -466,6 +466,34 @@ func (m *Mender) TransitionState(to State, ctx *StateContext) (State, bool) {
 	return transitionState(to, ctx, m)
 }
 
+func spinEventLoop(c *ControlMapPool, ctx *StateContext, to State) (State, bool) {
+	for {
+		switch to.Transition() {
+		case ToArtifactReboot_Enter, ToArtifactCommit_Enter, ToArtifactInstall:
```

Comment:
It's important that this handling only happens at `ArtifactInstall_Enter`. Does this happen at every internal transition within `ArtifactInstall`, including Leave? I don't remember exactly how these mechanics work, but this should be checked. It's possible we have to create a new const for the individual `Enter` and `Leave` states.

Reply:
It looks like `ToArtifactInstall` is enough, considering this only accepts the `to State` transition :)